pkg/api/v1alpha1/nutanixdatacenterconfig_webhook.go

```diff
@@ -0,0 +1,58 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// nutanixdatacenterconfiglog is for logging in this package.
+var nutanixdatacenterconfiglog = logf.Log.WithName("nutanixdatacenterconfig-resource")
+
+// SetupWebhookWithManager sets up the webhook with the manager.
+func (r *NutanixDatacenterConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-anywhere-eks-amazonaws-com-v1alpha1-nutanixdatacenterconfig,mutating=false,failurePolicy=fail,sideEffects=None,groups=anywhere.eks.amazonaws.com,resources=nutanixdatacenterconfigs,verbs=create;update,versions=v1alpha1,name=validation.nutanixdatacenterconfig.anywhere.amazonaws.com,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Validator = &NutanixDatacenterConfig{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
+func (r *NutanixDatacenterConfig) ValidateCreate() error {
+	nutanixdatacenterconfiglog.Info("validate create", "name", r.Name)
+	if r.Spec.CredentialRef == nil {
+		return fmt.Errorf("credentialRef is required to be set to create a new NutanixDatacenterConfig")
+	}
+
```

Comment:
Can we add checks for whether reconcile is pause or not, similar to here https://github.com/aws/eks-anywhere/blob/5057edc20a854a190c7f6f5208e4dbdf985d4410/pkg/api/v1alpha1/vspheredatacenterconfig_webhook.go#L68-L71